<!-- mobile_manipulation_teleop.launch -->
<launch>
    <node name="joy"
          pkg="joy"
          type="joy_node" output="screen">
      <param name="dev" value="/dev/input/js0"/>
      <param name="autorepeat_rate" value="20"/>  
      <param name="coalesce_interval" value="0.02"/>      
    </node>

    <!-- Load parameters from the gripper configuration file -->
    <rosparam file="$(find mobile_manipulation_teleop)/config/gripper_config.yaml" command="load"/>

    <!-- Launch the GripperController node -->
    <node name="gripper_server" pkg="mobile_manipulation_teleop" type="gripper_server.py" output="screen">
        <param name="config_file" value="$(find mobile_manipulation_teleop)/config/gripper_config.yaml"/>
    </node>  

    <node name="vision_to_speech" pkg="mobile_manipulation_teleop" type="vision_to_speech.py" output="screen"/>

    <node name="mobile_manipulation_teleop" pkg="mobile_manipulation_teleop" type="mobile_manipulation_teleop.py" output="screen">
        <param name="config_file" value="$(find mobile_manipulation_teleop)/config/PS3.yaml"/>
    </node>

    <include file="$(find map_interaction)/launch/map_interaction.launch"/>
    <node name="battery_status" pkg="mobile_manipulation_teleop" type="battery_status.py" output="screen"/>
</launch>
